index.js

```diff
@@ -195,6 +195,14 @@ function xgettext(input, options, cb) {
     if (options['files-from']) {
       input = fs.readFileSync(options['files-from'], options['from-code'])
         .split('\n')
```

Comment:
The combination of this line and the lines you added feels a bit like a mix of 2 solutions.

I'd choose either of these:
* If you want to follow GNU gettext as close as possible, use [readline](https://nodejs.org/api/readline.html#readline_example_read_file_stream_line_by_line) to do the line splitting. After that you can use your code to do the trimming (\t \r \n and space). Ideally a future commit/PR can take care of ignoring lines that start with # (comments).
* Or use `.split(/\r\n|\r|\n/)` to split immediately and forget about \t and space.

To get the core functionality you were after, I'm perfectly fine with the second option. Of course getting close to GNU gettext is the goal, but this would already be a step in that direction.